### ISSUE TITLE
[FEA] ClusterfuzzLite integration

### ISSUE THREAD

**Issue by AdamKorcz** [rapidsai/cudf#11413]
**Is your feature request related to a problem? Please describe.**
cuDF has an extensive fuzzing suite that could benefit the project by running in the CI.

**Describe the solution you'd like**
In this issue I suggest [ClusterfuzzLite](https://google.github.io/clusterfuzzlite/) integration for cuDF. This would require:

1. Adding a `.github/workflows/cflite.yml` file.
2. Adding a .clusterfuzzlite directory with build files.
3. Rewriting the fuzzers into coverage-guided fuzzers via [Atheris](https://github.com/google/atheris).

**Describe alternatives you've considered**

**Additional context**
ClusterfuzzLite handles the management of running fuzzers in the CI when PRs are made. It has a number of features that are useful for projects with multiple fuzzers:

1. CFL reuses corpus so that the fuzzers don't start from scratch every run.
2. [Batch fuzzing](https://google.github.io/clusterfuzzlite/running-clusterfuzzlite/github-actions/#batch-fuzzing) will run the fuzzers periodically to look for harder-to-find bugs and build of the corpus.
3. Only fuzzers that cover code that is changed in PRs run in the CI.

I will be glad to take the lead on this one if it is of interest to cuDF.

**Comment by shwina** [comment]
Hi @AdamKorcz - thanks for reaching out! We're definitely interested in improvements around our fuzzing infrastructure and appreciate any help!

We're currently in the process of migrating our CI to use GitHub Actions and we expect that to be complete in the coming weeks. I imagine that will make it easier to integrate a tool like ClusterfuzzLite. Perhaps we can touch base again once that migration is complete?

**Comment by vyasr** [comment]
Closing in favor of #16129 as a tracking issue.
